cosmic: geoclue2 fixes

thefossguy opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on nixos/modules/services/desktop-managers/cosmic.nix:
```diff
@@ -141,6 +141,7 @@ in
     services.accounts-daemon.enable = true;
     services.displayManager.sessionPackages = [ pkgs.cosmic-session ];
     services.geoclue2.enable = true;
+    services.geoclue2.agentWhitelist = [ "geoclue-demo-agent" ];
     services.geoclue2.enableDemoAgent = false;
```
Could you add that to the comment? Don't let others guess.

thefossguy replied:
Sorry, I should've already done that. I added the reasoning for my changes in the commit modifying the geoclue2 module. Just force pushed them.

Aleksanaa replied:
I think you still need a comment here above these two lines. People would not like to see changes here -> navigate to geoclue2 module -> check git blame -> see relevant commit message, or see changes here -> navigate to this PR -> check another commit -> see relevant commit message.

thefossguy replied:
Just to clarify, you are suggesting I add comments explaining the change in the cosmic module, right?

Aleksanaa replied:
I'm suggesting you add comments to clarify why you have to set `services.geoclue2.enableDemoAgent = false;` while adding `services.geoclue2.agentWhitelist = [ "geoclue-demo-agent" ];` in the cosmic module.

thefossguy replied:
Addressed in the latest force push :)

thefossguy committed:
nixos/geoclue2: add option to manage whitelisted agents

The geoclue2 module recommends that the guest agent be disabled when the
desktop environment provides their own geoclue2 agent. But when a
desktop environment uses the demo agent directly, like COSMIC does, the
demo agent must be whitelisted. But disabling the demo agent also
removes it from the whitelisted agents.

This commit adds an option which holds a list of all whitelisted
geoclue2 agents. It allows for consumers like COSMIC to have the demo
agent disabled but still whitelisted for such use cases.

thefossguy committed:
cosmic/nixos: whitelist `geoclue-demo-agent`

Aleksanaa commented:
nit: [cosmic/nixos: whitelist geoclue-demo-agent](https://github.com/NixOS/nixpkgs/pull/415901/commits/358d8a18bd778f688f6dea76f41afc2f5f5df031)

Aleksanaa commented:
nit is the short for nitpick, which means suggesting a change that's not important in the context. Here I wanted to say the commit message is wrong, that's not a big problem, but you should be careful next time.

thefossguy commented:
Can you point out what the correct commit message should've been? I'll keep that in mind next time. Thank you :)
